packages/rum-recorder/src/domain/rrweb/mutationBatch.ts

```diff
@@ -0,0 +1,54 @@
+import { monitor, noop } from '@datadog/browser-core'
+
+/**
+ * Maximum duration to wait before processing mutations. If the browser is idle, mutations will be
+ * processed more quickly. If the browser is busy executing small tasks (ex: rendering frames), the
+ * mutations will wait MUTATION_PROCESS_MAX_DELAY milliseconds before being processed. If the
+ * browser is busy executing a longer task, mutations will be processed after this task.
+ */
+const MUTATION_PROCESS_MAX_DELAY = 100
+
+export function createMutationBatch(processMutationBatch: (mutations: MutationRecord[]) => void) {
+  let cancelScheduledFlush = noop
+  const pendingMutations: MutationRecord[] = []
+
+  function flush() {
+    cancelScheduledFlush()
+    processMutationBatch(pendingMutations.splice(0, pendingMutations.length))
+  }
```

Comment:
what about clarify the intention by avoiding splice?
```suggestion
  let pendingMutations: MutationRecord[] = []

  function flush() {
    cancelScheduledFlush()
    processMutationBatch(pendingMutations)
    pendingMutations = []
  }
```